src/lair/lair_chip.rs

```diff
@@ -78,7 +83,7 @@ impl<'a, F: PrimeField32, H: Chipset<F>> MachineAir<F> for LairChip<'a, F, H> {
 
     fn name(&self) -> String {
         match self {
-            Self::Func(func_chip) => format!("Func[{}]", func_chip.func.name),
+            Self::Func { func_chip, .. } => format!("Func[{}]", func_chip.func.name),
```

Comment:
I’m not sure if this is checked by the current verifier, but I think there might be a risk of collision if we include the same chip multiple times.